util.bzl

```diff
@@ -31,9 +31,13 @@ def hazel_hackage(name, version, sha, **kwargs):
     return [(name, {"version": version, "sha256": sha} + kwargs)]
 
 # Things we override from GitHub
-def hazel_github_external(project, name, commit, sha):
-    return [(name, {"url": "https://github.com/" + project + "/" + name + "/archive/" + commit + ".zip", "sha256": sha, "stripPrefix": name + "-" + commit})]
+def hazel_github_external(project, repoName, commit, sha, directory="", name=None):
+    return [(name or repoName,
+             {"url": "https://github.com/" + project + "/" + repoName + "/archive/" + commit + ".zip",
+              "sha256": sha,
+              "stripPrefix": repoName + "-" + commit + directory
+             })]
 
 # Things we get from the digital-asset GitHub
-def hazel_github(name, commit, sha):
-    return hazel_github_external("digital-asset", name, commit, sha)
+def hazel_github(repoName, commit, sha, directory="", name=None):
+    return hazel_github_external("digital-asset", repoName, commit, sha, directory=directory, name=name)
```

Comment:
Is there no nicer way in Python of keeping the names optional and only passing those which are specified? Avoiding repeating the defaults twice?

Reply:
Yeah, I can add **kwargs to the next function, it might just end up pulling unexpected values into scope. I think maybe keeping the bazel functions simple might be a good idea as not everyone editing them is au fait with python

Reply:
Yeah, agreed - let's keep it simple.